treeherder/client/thclient/client.py

```diff
@@ -552,6 +553,25 @@ def validate(self):
         for d in self.data:
             d.validate()
 
+    def get_chunks(self, chunk_size):
+        """
+        Return a generator of new collections broken into chunks of size ``chunk_size``.
+
+        Each chunk will be a ``TreeherderCollection`` of the same
+        type as the original with a max of ``chunk_size`` count of
+        ``TreeherderData`` objects.
+
+        Each collection must then be POSTed individually.
+        """
+        for i in range(0, len(self.data), chunk_size):
+            # we must copy not only the data chunk,
+            # but also the endpoint_base.  In the case of
+            # a TreeherderJobCollection, this is passed in
+            # to the constructor.
+            chunk = copy.copy(self)
```

Comment:
This is the crux of the fix.  This ensures we get the ``endpoint_base`` and any other values the original collection had set.